### PR TITLE
Add accurate postal code

### PR DESCRIPTION
**Pull request by gauthierm**
Adds reverse geocode lookup of postal code to get more accurate codes.

Fixes partial postal code matches for the following addresses:

 - 355 Saint Peters Road, PE — `C1C` — (but “395 Saint Peters Road” works)
 - 84 Fitzroy Street, PE — `C1A` — (but “83 Fitzroy Street” works)
 - 395 Saint Peters Road, Morel Siding, NB — `E3Z`

### CODE REVIEW

File: src/AccessibleGooglePlacesAutocomplete.tsx [hunk 1]
```diff
@@ -227,4 +254,56 @@ export class AccessibleGooglePlacesAutocomplete extends React.Component<
...
+      this.geocoderService!.geocode(
+        {
+          location: place.geometry.location
+        },
+        (
+          geocodeResult: google.maps.GeocoderResult[],
+          geocodeStatus: google.maps.GeocoderStatus
+        ) => {
+          if (geocodeStatus === google.maps.GeocoderStatus.OK) {
+            resolve(geocodeResult[0]); // TODO get best match
```
Comment (gauthierm) [inline src/AccessibleGooglePlacesAutocomplete.tsx]: Needs to be done before merge.

File: src/AccessibleGooglePlacesAutocomplete.tsx [hunk 2]
```diff
@@ -101,29 +104,51 @@ export class AccessibleGooglePlacesAutocomplete extends React.Component<
...
+
+        if (useMoreAccuratePostalCode) {
+          const geocodeResult = await this.getReverseGeocodeData(placeResult);
+
+          // Remove Google Places postal code that may be a partial code.
+          placeResult.address_components = placeResult.address_components.filter(
+            component => !component.types.includes('postal_code')
+          );
+
+          console.log(placeResult.address_components); // tslint:disable-line
```
Comment (nburka) [inline src/AccessibleGooglePlacesAutocomplete.tsx]: remove `console.log` before merge

File: src/AccessibleGooglePlacesAutocomplete.tsx [hunk 3]
```diff
@@ -101,29 +104,51 @@ export class AccessibleGooglePlacesAutocomplete extends React.Component<
...
-          placeResult: google.maps.places.PlaceResult,
-          requestStatus: string
-        ) => {
-          if (requestStatus === google.maps.places.PlacesServiceStatus.OK) {
-            this.props.onConfirm(placeResult);
-          }
+      try {
+        const placeResult = await this.getPlaceDetails(selectedPrediction);
+
+        if (useMoreAccuratePostalCode) {
```
Comment (nburka) [inline src/AccessibleGooglePlacesAutocomplete.tsx]: it'd be nice if we could just do this _if_ the postal code wasn't already accurate. But I realize that would involve some custom checking like `if ((countryId === 'CA' && postalCode.length < 6) || (countryId === 'US' && postalCode.length < 5))` (pseudo-code). But it'd be faster/better/cheaper to use the correct postal code if returned the first time, which works in 90%+ cases.

**Review by nburka — CHANGES_REQUESTED** [review]

File: src/AccessibleGooglePlacesAutocomplete.tsx (hunk 3 above)
Comment (gauthierm) [reply src/AccessibleGooglePlacesAutocomplete.tsx]: We can check if the original field has the type `postal_code_prefix` and then do the more accurate lookup.

**Pull request merged**